ibis/backends/base/sql/registry/window.py

```diff
@@ -42,201 +27,140 @@
 }
 
 
-def _replace_interval_with_scalar(op: ops.Value) -> float | ir.FloatingScalar:
-    """Replace an interval type or expression with its equivalent numeric scalar.
-
-    Parameters
-    ----------
-    op
-        float or interval expression.
-        For example, `ibis.interval(days=1) + ibis.interval(hours=5)`
-
-    Returns
-    -------
-    preceding
-        `float` or `ir.FloatingScalar`, depending on the expr.
-    """
-    if isinstance(op, ops.Literal):
-        unit = getattr(op.output_dtype, "unit", "us")
-        try:
-            micros = _map_interval_to_microseconds[unit]
-            return op.value * micros
-        except KeyError:
-            raise ValueError(f"Unsupported unit {unit!r}")
-    elif op.args and isinstance(op.output_dtype, dt.Interval):
-        if len(op.args) > 2:
-            raise NotImplementedError("'preceding' argument cannot be parsed.")
-        left_arg = _replace_interval_with_scalar(op.args[0])
-        right_arg = _replace_interval_with_scalar(op.args[1])
-        method = _map_interval_op_to_op[type(op)]
-        return method(left_arg, right_arg)
+def cumulative_to_window(translator, func, frame):
+    klass = _cumulative_to_reduction[type(func)]
+    func = klass(*func.args)
+
+    try:
+        rule = translator._rewrites[type(func)]
+    except KeyError:
+        pass
     else:
-        raise TypeError(f'input has unknown type {type(op)}')
+        func = rule(func)
+
+    frame = frame.copy(start=None, end=0)
+    expr = an.windowize_function(func.to_expr(), frame)
+    return expr.op()
 
 
-def cumulative_to_window(translator, op, window):
-    klass = _cumulative_to_reduction[type(op)]
-    new_op = klass(*op.args)
+def interval_boundary_to_integer(boundary):
```

Comment:
Looks like this function is missing some test coverage. Can we delete the uncovered code or add a test?

Reply:
Covered by the bigquery tests.